alvr/server_openvr/src/lib.rs

```diff
@@ -114,22 +114,36 @@ fn event_loop(events_receiver: mpsc::Receiver<ServerCoreEvent>) {
                             FfiDeviceMotion::default()
                         };
 
-                        let ffi_left_controller_motion = context
-                            .get_device_motion(*HAND_LEFT_ID, poll_timestamp)
-                            .map(|motion| {
-                                let motion =
-                                    motion.predict(poll_timestamp, target_controller_timestamp);
-                                tracking::to_ffi_motion(*HAND_LEFT_ID, motion)
-                            })
-                            .filter(|_| tracked);
-                        let ffi_right_controller_motion = context
-                            .get_device_motion(*HAND_RIGHT_ID, poll_timestamp)
-                            .map(|motion| {
-                                let motion =
-                                    motion.predict(poll_timestamp, target_controller_timestamp);
-                                tracking::to_ffi_motion(*HAND_RIGHT_ID, motion)
-                            })
-                            .filter(|_| tracked);
+                        let mut ffi_left_controller_motion = None;
+                        let mut ffi_right_controller_motion = None;
+                        if let Some(config) = &controllers_config {
```

Comment:
I'd honestly prefer this being `(ffi_left_controller_motion, right...) = if let ...  () else (None,None)`. And then you can also remove the `tracked` filter and slap it into the if condition.